### ISSUE TITLE
python vs pythonw headache

### ISSUE THREAD

**Issue by moonshoes87** [Caoimhinmg/PmagPy#2]
In Python 2, we currently run all GUIs with #!/usr/bin/env pythonw, which is required for Anaconda and works for other distributions of Python.

The problem is, not all versions of Python 3 include pythonw.  Continuing to use the same shebang line means that some version of pythonw from the $PATH will be grabbed, 3.0 or no.  Since OS X comes with Python 2 installed by default, you are going to get a pythonw 2.7 if you don't have a pythonw 3.0, which then results in a lot of syntax errors and everything breaks.  

Now, there are a few possible ways for making a conditional shebang line, but it isn't pretty.  

You can see some info about this in these stack overflow questions.  

One very [elegant solution](http://stackoverflow.com/questions/18993438/shebang-env-preferred-python-version).  Unfortunately our situation is a bit too complicated to implement in this way, as far as I can tell.

A solution that is more [customizable and usable in our case](http://stackoverflow.com/questions/12070516/conditional-shebang-line-for-different-versions-of-python)

I've written a wrapper script that first checks if there is a pythonw that is 3.0.  If so, the script will use that pythonw to run the script name and args that follow.  If not, the script looks for a python that is 3.0., and tries to use that.  Usage looks like this:

    wrapper.sh magic_gui.py -DM 3

It works with both Anaconda and with EDM Python 3.  I don't love this solution, but I'm not sure if there is a better way.  

Thoughts?

**Comment by Caoimhinmg** [comment]
You could possibly do the conditional sh bang option by using python -V to check python version. Something like:

if pythonw and 3 in pythonw -V then exec pythonw
if python and 3 in python -V then exec python

Trouble is trying to do that with shell code which is not ideal for such one line conditionals. This might be more trouble than it is worth though, and I think the wrapper a perfectly good solution for the time being.

**Comment by moonshoes87** [comment]
I'll try this out, thanks.

**Comment by moonshoes87** [comment]
Right, so the problem remains with the elegant solution that I can't figure out how to write it simply enough and without internal quotations so that it can live in a python script.  I'm not that experienced with bash, unfortunately.  

@Caoimhinmg, any ideas about syntax for the above suggestion to make it fit?  Otherwise the wrapper may have to do.

**Comment by Caoimhinmg** [comment]
I'm **okay** at bash at best, I mostly just use it to move large amounts of stuff around my file system. There is no way my bash voodoo is good enough for this. I can go see what the different Linux forums have on this kind of thing, but it might take more time than it's worth. I think for now we let this rest and re-examine this at the end if we can solve the other problems.

**Comment by moonshoes87** [comment]
Ok, I'll consider this on hold.
